test/jdk/java/foreign/TestSegments.java

```diff
@@ -219,4 +237,51 @@ static Object defaultValue(Class<?> c) {
             return null;
         }
     }
+
+    @DataProvider(name = "accessModes")
+    public Object[][] accessModes() {
+        int nActions = AccessActions.values().length;
+        Object[][] results = new Object[nActions * nActions][];
```

Comment:
I think this should be:
```suggestion
        Object[][] results = new Object[1 << nActions][];
```
Since the number of combinations is 2^4. Though for 4 access modes 4 * 4 and 1 << 4 are both 16, it becomes a problem if we add or remove access modes.

Reply:
whoops - good catch